From c3bed2f6e7d2cff0bcd363102f87b6582387fd95 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 15 Feb 2018 09:11:32 -0800
Subject: [PATCH] TOOLS-1986 MG's ./tools/ls-missing-release-builds will
 falsely report missing builds if mls env isn't properly setup

---
 tools/ls-missing-release-builds | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/ls-missing-release-builds b/tools/ls-missing-release-builds
index 809dc9a..3b1d3f0 100755
--- a/tools/ls-missing-release-builds
+++ b/tools/ls-missing-release-builds
@@ -71,6 +71,10 @@ if [[ -z "$targets" ]]; then
         | grep -v '^headnode-joyent-debug ')
 fi
 
+# Ensure mls is setup properly at all.
+mls /Joyent_Dev/public/builds >/dev/null \
+    || fatal "cannot list /Joyent_Dev/public/builds"
+
 echo "$targets" | while read target public; do
     if [[ "$public" == "true" ]]; then
         latest_mpath=/Joyent_Dev/public/builds/$target/$release-latest
-- 
2.21.0

